server/index.js

```diff
@@ -316,6 +325,305 @@ app.get('/api/fighters/data/:type', (req, res, next) => {
+  let { fighterId } = req.body;
+  fighterId = Number(fighterId);
+  const fullResult = {};
+  if (!fighterId) {
+    throw new ClientError(400, 'fighterId must be a number');
+    return;
+  }
+  if (!checkValidity([name, moveType, damage, activeFrames, totalFrames])) {
+    throw new ClientError(400, 'must have (fighterId), (name), (moveType), (damage), (activeFrames), and (totalFrames) as parameters');
+    return;
+  }
+  if (typeof fighterId === 'number'
+    & typeof name === 'string'
+    & typeof moveType === 'string'
+    & typeof damage === 'string'
+    & typeof activeFrames === 'string'
+    & typeof totalFrames === 'string') {
+      return handleMoves(1)
+      function handleMoves (counter) {
+        if (counter === 1) {
+          const sql = `
+          INSERT INTO public.moves
+          ("fighterId", "name", "moveType", "type")
+          VALUES ($1, $2, $3, 'move')
+          RETURNING *;
+          `;
+          const params = [fighterId, name, moveType];
+          return db.query(sql, params)
+            .then(result => {
+              Object.assign(fullResult, result.rows[0]);
+              return handleMoves(counter + 1)
+            })
+            .catch(err => {
+              if (/already exists/g.test(err.detail)
+              || /not present/g.test(err.detail)) {
+                throw new ClientError(400, `${err.detail}` );
+              } else {
+                next(err);
+              }
+            });
+        } else {
+          const sql2 = `
+            INSERT INTO public.hitboxes
+              ("damage", "activeFrames", "totalFrames")
+            VALUES ($1, $2, $3)
+            RETURNING *;
+          `;
+          const params2 = [damage, activeFrames, totalFrames];
+          return db.query(sql2, params2)
+            .then(result => {
+              Object.assign(fullResult, result.rows[0]);
+              res.status(201).json(fullResult);
+            })
+            .catch(err => {
+              if ((/already exists/g.test(err.detail))
+              || (/not present/g.test(err.detail))) {
+                throw new ClientError(400, `${err.detail}`);
+              } else {
+                next(err);
+              }
+            });
+        }
+      }
+  }
+});
+
+app.post('/api/add/throws', (req, res, next) => {
+  const { name, damage, activeFrames, totalFrames } = req.body;
+  let { fighterId } = req.body;
+  fighterId = Number(fighterId);
+  const fullResult = {};
+  if (!fighterId) {
+    throw new ClientError(400, 'fighterId must be a number');
+    return;
+  }
+  if (!checkValidity([name, damage, activeFrames, totalFrames])) {
+    throw new ClientError(400, 'must have (fighterId), (name), (damage), (activeFrames), and (totalFrames) as parameters');
+    return;
+  }
+  if (typeof fighterId === 'number'
+    & typeof name === 'string'
+    & typeof damage === 'string'
+    & typeof activeFrames === 'string'
+    & typeof totalFrames === 'string') {
```

Comment:
What about your `else` here? If the server doesn't respond, the client will hang and the server will leak socket connections. This applies to all routes. There can't be any logic paths left over that don't result in the server responding to the client.